### PR TITLE
Micro Image Case List Format

### PR DESCRIPTION
**Pull request by zandre-eng**
## Product Description

Adding the `Image` format to a case list configuration:
image-format

Case list in mobile using micro images:
ca5fc92c-9b2a-4ea6-b07e-252d56244139_resized

## Technical Summary

Link to ticket [here](https://dimagi.atlassian.net/browse/SC-3405).

A new `Image` format was added to the case list configuration, allowing users to easily set up showing micro images in the mobile case list. Each case list configuration can only have one micro image column present, and the property name for this format cannot be changed (defaults to the reserved `cc_case_image` property name for micro images).

To have the `Image` format be available, the `VELLUM_CASE_MICRO_IMAGE` feature flag needs to be enabled.

For more context, please refer to [this](https://docs.google.com/document/d/14yDNK6RyQGB-8H6DioGgoj0c2mwJuI8nWck4zFMYFAs/edit?pli=1#heading=h.cov9a7u3rjtf) tech spec document.

## Feature Flag

`VELLUM_CASE_MICRO_IMAGE`

### Safety story

- Local testing
- Testing on staging

Changes only affect the newly added `Image` format type. These changes would therefore only affect domains that have the relevant feature flag enabled.

### Automated test coverage

N/A

### QA Plan

No QA planned.

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by Charl1996 — APPROVED** [review]
I'm OK with these changes, but I'll let someone else with more knowledge around the app_manager have the final say.

**Review by mkangia — APPROVED** [review]
Looks good.

Are there any documentation updates being done for this? I believe internal for now since its FF

**Comment by zandre-eng** [comment]
> Are there any documentation updates being done for this? I believe internal for now since its FF

We do have some internal documentation, however this still explains using case tiles and will need to be updated to include the changes from this PR. I'll create a separate ticket for updating this.

File: corehq/apps/app_manager/static/app_manager/js/details/column.js [hunk 1]
```diff
@@ -364,6 +365,13 @@ hqDefine("app_manager/js/details/column", function () {
         self.$format = $('<div/>').append(self.format.ui);
         self.$format.find("select").css("margin-bottom", "5px");
         self.format.on('change', function () {
+            if (self.field.val() === microCaseImageName && self.format.val() !== 'image') {
```
[the remark this answers is not in the document]
Comment (zandre-eng) [reply corehq/apps/app_manager/static/app_manager/js/details/column.js:368]: Comment has been added in [53f53fb](https://github.com/dimagi/commcare-hq/pull/34766/commits/53f53fb7006c74e91cd7bca497e63873ae54edb4). Please let me know if it gives a good enough explanation.

File: corehq/apps/app_manager/static/app_manager/js/details/column.js [hunk 2]
```diff
@@ -426,6 +434,11 @@ hqDefine("app_manager/js/details/column", function () {
                         self.time_ago_extra.value = interval.val();
                         fireChange();
                     });
+                } else if (this.val() === 'image') {
```
[the remark this answers is not in the document]
Comment (zandre-eng) [reply corehq/apps/app_manager/static/app_manager/js/details/column.js:439]: Also added a comment here in [53f53fb](https://github.com/dimagi/commcare-hq/pull/34766/commits/53f53fb7006c74e91cd7bca497e63873ae54edb4)

File: corehq/apps/app_manager/static/app_manager/js/details/column.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (mkangia) [reply corehq/apps/app_manager/static/app_manager/js/details/column.js:368]: nit: I think for me what was confusing/unclear was why we are doing this reset. I could understand from code what was happening but not the why, which was not very obvious.
So, the comment would be what you were thinking, that made you write that code?

What I had in mind was 
> for micro case image field, if format is changed to anything other than image
HQ should reset the field and make it editable.

You can see how its different from
> After changing the format to something other than image, clear the reserved micro image field
name and make the input editable again

Its a minor difference but its useful for reader

File: corehq/apps/app_manager/static/app_manager/js/details/column.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (mkangia) [reply corehq/apps/app_manager/static/app_manager/js/details/column.js:439]: Similarly here could be

> if format is set to image fix field to micro case image

File: corehq/apps/app_manager/static/app_manager/js/details/column.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (zandre-eng) [reply corehq/apps/app_manager/static/app_manager/js/details/column.js:368]: @mkangia Ah, I understand now. How does the following comment sound?

> The field name input was disabled to enforce using the reserved micro image name. If the format is no longer an image then the user can edit the field input again

We disable the input to enforce the field name that is used for the micro image. Users should only be using `cc_case_image` to store the micro image, so we simply force that name usage here by disabling the input.

File: corehq/apps/app_manager/static/app_manager/js/details/column.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (zandre-eng) [reply corehq/apps/app_manager/static/app_manager/js/details/column.js:439]: @mkangia How does the following comment sound?

> We are enforcing the reserved field name for the micro image format, so don't allow a user to change this

This gives a bit more of a reasoning as to why the input is being disabled.

File: corehq/apps/app_manager/static/app_manager/js/details/column.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (mkangia) [reply corehq/apps/app_manager/static/app_manager/js/details/column.js:368]: that is even better.

File: corehq/apps/app_manager/static/app_manager/js/details/column.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (mkangia) [reply corehq/apps/app_manager/static/app_manager/js/details/column.js:439]: works for me.

File: corehq/apps/app_manager/static/app_manager/js/details/column.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (zandre-eng) [reply corehq/apps/app_manager/static/app_manager/js/details/column.js:368]: Cool, I've updated the comment text in [3f8d6e0](https://github.com/dimagi/commcare-hq/pull/34766/commits/3f8d6e0ff3215ea2c8f398e13fd87d5d47b7d04c)

File: corehq/apps/app_manager/static/app_manager/js/details/column.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (zandre-eng) [reply corehq/apps/app_manager/static/app_manager/js/details/column.js:439]: Comment text has been updated in [3f8d6e0](https://github.com/dimagi/commcare-hq/pull/34766/commits/3f8d6e0ff3215ea2c8f398e13fd87d5d47b7d04c)

**Review by mkangia — APPROVED** [review]

**Pull request merged**